bubblegum/program/src/lib.rs

```diff
@@ -516,6 +547,99 @@ fn process_mint_v1<'info>(
     )
 }
 
+fn process_creator_verification<'info>(
+    ctx: Context<'_, '_, '_, 'info, CreatorVerification<'info>>,
+    root: [u8; 32],
+    data_hash: [u8; 32],
+    creator_hash: [u8; 32],
+    nonce: u64,
+    index: u32,
+    creators: Vec<Creator>,
+    verify: bool,
+) -> Result<()> {
+    let owner = ctx.accounts.owner.to_account_info();
+    let delegate = ctx.accounts.delegate.to_account_info();
+    let merkle_slab = ctx.accounts.merkle_slab.to_account_info();
+
+    let creator = ctx.accounts.owner.key();
+
+    // Creator Vec must contain creators.
+    if creators.is_empty() {
+        return Err(BubblegumError::NoCreatorsPresent.into());
+    }
+
+    // Creator must be in user-provided creator Vec.
+    if !creators.iter().any(|c| c.address == creator) {
+        return Err(BubblegumError::CreatorNotFound.into());
+    }
+
+    // User-provided creator Vec must result in same user-provided creator hash.
+    let provided_creator_data = creators
+        .iter()
+        .map(|c| [c.address.as_ref(), &[c.verified as u8], &[c.share]].concat())
+        .collect::<Vec<_>>();
+    let calculated_creator_hash = keccak::hashv(
+        provided_creator_data
+            .iter()
+            .map(|c| c.as_slice())
+            .collect::<Vec<&[u8]>>()
+            .as_ref(),
+    );
+    assert_eq!(creator_hash, calculated_creator_hash.to_bytes());
+
+    // Calculate new creator hash with `verified` set to true for signing creator.
+    let updated_creator_data = creators
+        .iter()
+        .map(|c| {
+            let verified = if c.address == creator.key() {
+                verify
+            } else {
+                c.verified
+            };
+            [c.address.as_ref(), &[verified as u8], &[c.share]].concat()
+        })
+        .collect::<Vec<_>>();
+    let updated_creator_hash = keccak::hashv(
+        updated_creator_data
+            .iter()
+            .map(|c| c.as_slice())
+            .collect::<Vec<&[u8]>>()
+            .as_ref(),
+    );
+
+    let asset_id = get_asset_id(&merkle_slab.key(), nonce);
+    let previous_leaf = LeafSchema::new_v0(
+        asset_id,
+        owner.key(),
+        delegate.key(),
+        nonce,
+        data_hash,
```

Comment:
TODO: I need to rehash data after updating creator array, because this hash also contains creator array.  This change in progress now.

Reply:
Fixed in next version